codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/protocols/serialize/SerializerUtil.kt

```diff
@@ -6,28 +6,17 @@
 package software.amazon.smithy.rust.codegen.core.smithy.protocols.serialize
 
 import software.amazon.smithy.model.Model
-import software.amazon.smithy.model.shapes.BooleanShape
-import software.amazon.smithy.model.shapes.DoubleShape
-import software.amazon.smithy.model.shapes.FloatShape
 import software.amazon.smithy.model.shapes.MemberShape
-import software.amazon.smithy.model.shapes.NumberShape
 import software.amazon.smithy.model.shapes.StructureShape
 import software.amazon.smithy.rust.codegen.core.rustlang.RustWriter
 import software.amazon.smithy.rust.codegen.core.rustlang.Writable
 import software.amazon.smithy.rust.codegen.core.rustlang.rustBlock
 
 class SerializerUtil(private val model: Model) {
     fun RustWriter.ignoreZeroValues(shape: MemberShape, value: ValueExpression, inner: Writable) {
-        val expr = when (model.expectShape(shape.target)) {
```

Comment:
We need to revert this. This affects the client. We said we have to evaluate the impact of serializing zero values in the SDK. This is a place where behavior among clients and servers will be different: clients will implement presence-tracking, servers will always serialize everything.

Reply:
I've not changed the behaviour here - this is just a refactoring pending a decision on what we want to do going forward for zero values in the SDK.

Reply:
Some extra context:
We started [always serializing required query params](https://github.com/awslabs/smithy-rs/pull/1973) after an issue related to S3's multipart upload operations.